---
title: "Seurat Output"
date: "`r format(Sys.time(), '%m/%d/%Y')`"
output: 
  html_document:
    toc: true
    toc_float: true
---
## Introduction

This is literally a copy and paste of this [Seurat tutorial](http://satijalab.org/seurat/pbmc3k_tutorial.html) meant as a first pass at 10x data.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Setup the Seurat Object

The original tutorial analyzed a dataset of Peripheral Blood Mononuclear Cells (PBMC) freely available from 10X Genomics. There are 2,700 single cells that were sequenced on the Illumina NextSeq 500. The raw data can be found [here](https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz).

We start by reading in the data. All features in Seurat have been configured to work with sparse matrices which results in significant memory and speed savings for Drop-seq/inDrop/10x data.

```{r SeuratObject, echo = FALSE, message = FALSE, warning = FALSE}
library(Seurat)
library(dplyr)
library(Matrix)

# Load the dataset
raw_data <- Read10X( data.dir = {{path}} )

# Initialize the Seurat object with the raw (non-normalized data).  Keep all
# genes expressed in >= 3 cells (~0.1% of the data). Keep all cells with at
# least 200 detected genes
obj <- CreateSeuratObject(raw.data = raw_data, 
                          min.cells = {{cells}}, 
                          min.genes = {{genes}}, 
                          project = {{project}})
```


